### PR TITLE
[YUNIKORN-2847] Documentation: Update new label/annotation changes before 1.6.0 release

### PR DESCRIPTION
**Pull request by chenyulin0719**
### What is this PR for?

Add below descriptions:
1. Add new canonical representation labels: `yunikorn.apache.org/app-id`, `yunikorn.apache.org/queue`, also update to the  latest prioirty. Add note for inconsistent pod rejection in 1.7.0.
3. Mark the non-canonical representation as 'Classic' since we're not going to deprecate it.
4. Use the full annotation name in annotation tables. (ex: `task-groups` to `yunikorn.apache.org/task-groups` )
5. Update the description of `spark-app-selector`. Add a link to SparkOperator.

### What type of PR is it?
* [x] - Improvement

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2847

### Questions:
NA

### CODE REVIEW

File: docs/user_guide/labels_and_annotations_in_yunikorn.md [hunk 1]
```diff
@@ -22,33 +22,37 @@ under the License.
...
+|-------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [Classic]<br/>`yunikorn.apache.org/app-id`                  | It is equivalent to the label `yunikorn.apache.org/app-id`, and we recommend using it.                                                                                                                                                                                                                 |
+| [Classic]<br/>`yunikorn.apache.org/queue`                   | It is equivalent to the label `yunikorn.apache.org/queue`, and we recommend using it.                                                                                                                                                                                                                  |
```
Comment (craigcondit) [inline docs/user_guide/labels_and_annotations_in_yunikorn.md:46]: Let's switch from "[Classic] to [LEGACY], and reword descriptions as follows:

Equivalent to the preferred label `yunikorn.apache.org/queue`.

**Review by craigcondit — CHANGES_REQUESTED** [review]
In general, good. Some grammar / wording suggestions for clarity.

**Comment by chenyulin0719** [comment]
@craigcondit Thanks for the review.  Updated this PR with the suggestions.

**Review by craigcondit — APPROVED** [review]
+1 LGTM.

**Pull request closed without merging**